tutorials/pages/shimmerevm-dapp-voting.md

```diff
@@ -0,0 +1,328 @@
+    bool voted;  // if true, that person already voted
+    address delegate; // person delegated to
+    uint vote;   // index of the voted proposal
+}
+```
+
+Once you have created the `Voter`, you will need `Proposal` `struct` to nominate who to vote on:
+
+```solidity!
+mapping(address => Voter) public voters;
+```
+
+Once you have created the `Voter`, you will need `Proposal` `struct` to nominate who to vote on:
+
+```solidity!
+struct Proposal {
+    bytes32 name;   // short name (up to 32 bytes)
+    uint voteCount; // number of accumulated votes
+}
+```
+
+:::tip Gas Fees
+
+If you can, limit the length to a certain number of bytes. You must cover gas fees for any data you store on the chain if you use one of bytes1 to bytes32 as they are much cheaper. In other words, if you use short names, your gas fees will be lower.
+
+:::
+
+Next, you need to add a vote function, which will register the vote of every user toward a proposal. Let's start with the following:
+
+```solidity!
+function vote(uint proposal) public {
+    Voter storage sender = voters[msg.sender];
+}
+```
+
+As you need to check if the voter is allowed to vote, you should add the following:
+
+```solidity!
+require(sender.weight != 0, "Has no right to vote");
+```
+
+You also need to check if the voter has already voted, so you should also add the following conditional:
+
+```solidity!
+require(!sender.voted, "Already voted.");
+```
+
+If everything looks okay, register the vote:
+
+```solidity!
+sender.voted = true;
+sender.vote = proposal;
+proposals[proposal].voteCount += sender.weight;
+```
+
+#### Create functions to count the votes
+
+You will also need a function to count the votes and let the results be known. To do so, add a function called `winningProposal` which will return the index of the winning proposal:
+
+```solidity!
+function winningProposal() public view returns (uint winningProposal_) {
+    uint winningVoteCount = 0;
+    for (uint p = 0; p < proposals.length; p++) {
+        if (proposals[p].voteCount > winningVoteCount) {
+            winningVoteCount = proposals[p].voteCount;
+            winningProposal_ = p;
+        }
+    }
+}
+```
+
+To make things a little more human readable, add a function `winnerName` that returns the name of the winner:
+
+```solidity!
+function winnerName() public view returns (bytes32 winnerName_) {
+    winnerName_ = proposals[winningProposal()].name;
+}
+```
+
+At this point, the basic functionalities of the smart contract are taken care of. However, you could also need a few more functionalities, such as delegating votes and giving someone else the right to vote. I have added those in the final code, so feel free to keep or remove them as you see fit.
+
+Here's the complete smart contract code:
+
+```solidity!=
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+/**
+ * @title Voting
+ * @dev Implements voting process along with vote delegation
+ */
+contract Voting {
+
```

Comment:
Who do you want to target with this tutorial? 
If beginners then I guess you should also explain this part? If not ignore this comment :)